tests/unit/models/project-test.js

```diff
@@ -461,6 +461,16 @@ describe('models/project.js', function() {
       var addon = project.findAddonByName('not-an-addon');
       expect(addon).to.equal(undefined, 'not found addon should be undefined');
     });
+
+    it('should not return an addon that is a substring of requested name', function() {
+      var addon = project.findAddonByName('foo-ba');
+      expect(addon).to.equal(undefined, 'foo-ba should not be found');
+    })
+
+    it('should return an addon that is a suffix of requested name', function() {
+      var addon = project.findAddonByName('bar');
+      expect(addon).to.equal(undefined, 'should have found the foo-bar addon');
```

Comment:
I don't know that this is our desired behavior. I kinda want to `split('\')` and test the last segment. Is there a scenario where we want to just match the suffix?

In short:
(`bar` !== `foo-bar`)